.github/workflows/publish.yaml

```diff
@@ -0,0 +1,20 @@
+name: Publish to NPM
+on:
+  push:
+    branches:
+      - release
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+      - run: node --version
+      - run: npm install
+      - run: npm run rebuild-docker-image
+      - run: npm publish
```

Comment:
What's your advice for preventing releases where the tests fail?

The only "test" TravisCI ran before release was `npm run build`, and this will happen here as well since building is part of our `prepare` script. We already run some tests in `.github/workflows/ci.yaml`, but this publish action isn't dependent on that CI passing.

Reply:
Good point, I think the solution is to trigger on [`workflow_run`](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run) of the `ci` workflow. Implemented here: https://github.com/nextstrain/auspice/pull/1434/commits/595fd23cbe7ccba049b16a26d4487e7119a66938

Reply:
So reading a bit deeper, this stands out:

> This event will only trigger a workflow run if the workflow file is on the default branch.

I'm not sure if it means this won't run properly on `release` since it's not the default branch, but in our use cases `release` and `master` (default) will always be even at runtime, meaning the workflow file does exist on the default branch... only way to find out would be to do a test release (2.32.2?)

If this doesn't work, I'd suggest getting rid of `release` and instead triggering on the [GitHub release](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release) event which can be based on `master`.

Reply:
For `workflow_run` events, I believe their state is only ever `requested` or `completed`, with the latter including both successful and failed runs. That's not quite what we want here.

However, instead of having "publish" be a separate workflow, I suggest moving it into the "ci" workflow as a final job that conditions on the success of the test jobs and the current branch being `release`.

Reply:
@tsibley good point. Implemented here: https://github.com/nextstrain/auspice/pull/1434/commits/64268970677f990ebefd0f57e1510cefb7390414

Reference: [Contexts - GitHub Docs](https://docs.github.com/en/actions/learn-github-actions/contexts#determining-when-to-use-contexts)

There's even a nice visual on the [GitHub workflow summary page](https://github.com/nextstrain/auspice/actions/runs/1478161125):

<img src="https://user-images.githubusercontent.com/13424970/142489945-8f717d13-3b3e-4b5d-914b-1c0faacc11eb.png" alt="auspice CI graph" width="400" />